haskell/defs.bzl

```diff
@@ -554,6 +554,16 @@ haskell_import = rule(
             default = Label("@rules_haskell//haskell:version_macros"),
         ),
     },
+    doc = """\
+Import libraries that are prebuilt outside of Bazel.
+
+The attributes of this rule loosely correspond to the fields of the
+GHC package database. Refer to the [GHC User's Guide][ghc-doc-pkginfo]
+for documentation. If a library is registered in GHC's global package
+database, then you can use `haskell_toolchain_library` instead.
+
+[ghc-doc-pkginfo]: https://downloads.haskell.org/ghc/latest/docs/html/users_guide/packages.html#installedpackageinfo-a-package-specification
+""",
```

Comment:
The `haskell_import` rule only works for packages that are registered in the global package-db. It's not meant to be used by users directly. It's used to collect builtin packages and make them available to the toolchain. See https://github.com/tweag/rules_haskell/pull/925

Reply:
It sounds like the contract of `haskell_import` slightly changed over time. Let's say I have a GHC package that I built with my own means (i.e. not with `cabal_haskell_library` or `haskell_library`). How do I expose that to Bazel? Can I use `haskell_import` for that the way I can use `cc_import` to expose any odd C/C++ library along with some metadata?

Reply:
`haskell_import` assumes that the package is [listed in the global package db](https://github.com/tweag/rules_haskell/pull/859/files#diff-9b1fa0d6cbde73a03f524f92963b1b781de46bcd395e844ea3f3fc2ba5e10222R689), currently there is no way to import an additional package-db into Bazel. It is possible to import an externally built Haskell library as a C library via `cc_library` and use it as a dependency to regular Haskell targets. However, this won't include metadata like the package configuration.

The implementation of `haskell_import` is tailored towards toolchain packages and includes a few related workarounds. If we need a more generic `haskell_import` it may be better to rename the current one as `haskell_toolchain_library_import` or similar and implement the new `haskell_import` separately. AFAIK we didn't have a use-case so far that required a generic `haskell_import`.